### ISSUE TITLE
Add links are not working

### ISSUE THREAD

**Issue by collinbarrett** [collinbarrett/FilterLists#79]
An anonymous user reported that the "Add" links were not working on their setup. However, the links on the [old](https://v1.filterlists.com) site still work fine for them. Their setup is below. Is anyone else experiencing this? Any ideas on how to solve it?

Windows 10 x86
Chrome Version 56.0.2924.87
uBlock Origin 1.11.0

**Comment by collinbarrett** [comment]
The anonymous user who reported it is still experiencing it after the above attempts at fixing it. I am unable to reproduce, personally. If anyone else has this issue, please share your setup.

**Comment by Snapy** [comment]
@collinbarrett  We need setup _and_ **detailed steps**. Also we need to know if it works with ABP.

**Comment by collinbarrett** [comment]
Thanks, @Snapy . The user who reported it tried both the setup in the OP as well as Firefox Developer Edition on the same box. They sent me a video of them clicking a few of the add links on each of the following three domains in both browsers.

- https://adblockplus.org/subscriptions
- https://v1.filterlists.com
- https://filterlists.com

Add links on the first two domains successfully launched the Add popup for UBO in both browsers. The current FilterLists site failed, however. Note that this video was sent before https://github.com/collinbarrett/FilterLists/commit/2c4f6d67c8a20dd75ad0e10a06a8438dc2fbb884 , but they reported that it didn't work after that fix either. I don't have permissions to share the actual screengrabs they sent me. I can ask them if they could try ABP. What other information would be helpful that I could ask them? Possibly other plugins installed that might be conflicting (though I would think they would affect it on all three of the domains)?

It might be that I need to spend some time in UBO's source code to further understand how add links are parsed. And I need to side-by-side compare the urls on the three sites, as the html encoding does vary slightly. I'm just not sure the "correct"/preferred way to encode the urls for UBO and ABP. The current links on all three sites work fine for Chrome on Mac and WIn7 (my two daily setups).

**Comment by Snapy** [comment]
Here is the file handling abp: links : [subscriber.js](https://github.com/gorhill/uBlock/blob/master/src/js/scriptlets/subscriber.js). A screenshot of uBO dashboard could be useful (or any info from [here](https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md)) or error in the browser console. Otherwise you can ask gorhill for help.

**Comment by collinbarrett** [comment]
Ok, thanks for sharing that link. It looks like UBO just uses decodeURIComponent. ABP's [link generator tool](https://github.com/adblockplus/web.adblockplus.org/blob/a165911bcac73fe02b2c0f3558fd4df4072457e5/static/js/webtools.js) similarly uses encodeURIComponent and encodes the ampersands as well. I just deployed https://github.com/collinbarrett/FilterLists/commit/0eaa9e61c2e68bde87df73829a3bf7e0624d144b to try to match ABP's method (also opening #81 to look at later). I have notified the user who reported the issue to see if these tweaks solve it.

@gorhill, have you run across any issues with these abp subscribe links in the past? I imagine there could be browser/plugin conflicts that might disable this functionality on some setups?

Side Note: I am also aware that not all lists currently on FilterLists are 'subscribable' in UBO/ABP due to format and/or file type. I hope to address that at some point with #28 . But, the reporter of this issue was testing on popular/common lists that are 'subscribable', so that is not a relevant gotchya.

**Comment by collinbarrett** [comment]
Some further testing from the reporter:

He discovered two possible common links. In his tests:

| Windows | 7 | 10 |
|---|---|---|
| x86 | fail | fail |
| x64 | pass | pass |

But, also the x86 machines had a smaller viewport than the x64 machines, so he thought maybe screen-size could be the deciding factor as well? I don't know, neither screen size or processor architecture seems like it would influence this.

But, here were his/her final thoughts from last night:

> So I remoted to a Windows 7 x64 virtual machine (1920x1080) using my Windows 10 x86 device (1024x600) and got an error adding filters using uBlock Origin on Firefox. I was frustrated because previously I thought it was architecture-related. But then I asked the owner to log in and try it, it failed:
> 
> I instructed to maximize the window (forgot to refresh), it failed:
> 
> then asked to do it again but refresh the page before adding a filter and it worked!
> 
> Both W10 and W7 x86 devices have small screens and all the x64 devices it was tested on are standard-sized.

**Comment by gorhill** [comment]
uBO injects a content script, [subscriber.js](https://github.com/gorhill/uBlock/blob/master/src/js/scriptlets/subscriber.js), which content script, when injected, will try to find if there are any `abp:` link on the page. If none are found, it bails out, it does not leave any hooks behind, i.e. does not linger to watch for page changes to find out if `abp:` links are added afterward.

From the look of it, your new design causes the content of the page to load with a delay. This means it's possible that when `subscriber.js` is injected, there are no `abp:` link yet to be found.

**Comment by collinbarrett** [comment]
Ah. Thanks, @gorhill .

Yes, that makes a lot of sense. One of my big goals for the redesign was having a table that could be bound to json data (to open-source on GitHub) and to allow it to be sorted/filtered. But, the module I ended up with is starting to show its limitations.

For now, I just tried turning off a feature that it calls "Hide table until page is completely loaded". The page looks a bit uglier on loading, but it doesn't wait to load anything. Since all of the list information including the subscribe/add links are in the initial html response, it should all be available to the browser now in time for UBO's subscriber.js to catch them.

I'll check with the issue reporter to see if this improves it for them.

I'm going to continue to look for a better front-end module for displaying the data. Possibly something like [this](https://adazzle.github.io/react-data-grid/).
